### DISCUSSION TITLE
Why does the Linux AppImage connect to all these services upon startup? (including Google)

### DISCUSSION THREAD

**Discussion by ForeverNooob** [ztjhz/BetterChatGPT#575] [Q&A]
Hi,

I monitor my connections and I've noticed that `linux-better-chatgpt-1.0.5-x86_64.AppImage` was connecting to services like Github and Google (though the latter was only observed on first startup)

```
$ netstat -tupwn | grep -i 'better'
Active Internet connections (w/o servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 [redacted]              185.199.108.133:443     ESTABLISHED 3867948/better-chat
tcp        0      0 [redacted]              140.82.121.4:443        ESTABLISHED 3867948/better-chat
tcp6       0      0 [redacted]              ::1:51735               ESTABLISHED 3867948/better-chat
tcp6       0      0 [redacted]              ::1:59642               ESTABLISHED 3867489/better-chat
tcp6       0      0 [redacted]              ::1:59648               ESTABLISHED 3867489/better-chat
tcp6       0      0 [redacted]              ::1:59634               ESTABLISHED 3867489/better-chat
tcp6       0      0 [redacted]              ::1:51735               ESTABLISHED 3867948/better-chat
tcp6       0      0 [redacted]              ::1:51735               ESTABLISHED 3867948/better-chat
```

Why is that? I thought the client only ran locally and would only connect to outside services if you entered the API keys.
Are these analytics?

**Comment by ztjhz** [comment]
Hi @ForeverNooob, here’s an explanation of what is happening:

1. **GitHub Connections (185.199.108.133 and 140.82.121.4):** The desktop app uses some images which are hosted on GitHub. (An example can be found [here](https://github.com/ztjhz/BetterChatGPT/blob/main/src/components/AboutMenu/AboutMenu.tsx#L118))
2. **Google Connections:** BetterChatGPT uses the Google Drive API integration to synchronize chats and settings across different devices.
3. Localhost Connections (::1): The desktop app serves the BetterChatGPT frontend locally on localhost.

Hope this answers your question!
